cpp/ql/src/semmle/code/cpp/ir/implementation/raw/Instruction.qll

```diff
@@ -212,6 +218,7 @@ class Instruction extends Construction::TInstruction {
    * Gets the type of the result produced by this instruction. If the instruction does not produce
    * a result, its result type will be `IRVoidType`.
    */
+  cached
   final IRType getResultIRType() { result = getResultLanguageType().getIRType() }
```

Comment:
Caching the member predicate here means that three cached copies are produced even though only the `aliased_ssa` one is needed. The same goes for all the cached predicates in `unaliased_ssa`, of course. Let's remember to revisit that issue in the near future.

Reply:
github/codeql-c-analysis-team#98